benchmarks/src/main/java/org/opensearch/benchmark/routing/allocation/AllocationBenchmark.java

```diff
@@ -155,31 +185,96 @@ public void setUp() throws Exception {
             rb.addAsNew(metadata.index("test_" + i));
         }
         RoutingTable routingTable = rb.build();
-        DiscoveryNodes.Builder nb = DiscoveryNodes.builder();
-        for (int i = 1; i <= numNodes; i++) {
-            nb.add(Allocators.newNode("node" + i, Collections.singletonMap("tag", "tag_" + (i % numTags))));
-        }
         initialClusterState = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
             .metadata(metadata)
             .routingTable(routingTable)
-            .nodes(nb)
+            .nodes(setUpClusterNodes(sourceNodes, targetNodes))
             .build();
+        // Start all unassigned shards
+        initialClusterState = initialClusterStrategy.reroute(initialClusterState, "reroute");
+        while (initialClusterState.getRoutingNodes().hasUnassignedShards()) {
+            initialClusterState = initialClusterStrategy.applyStartedShards(
+                initialClusterState,
+                initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.INITIALIZING)
+            );
+            initialClusterState = initialClusterStrategy.reroute(initialClusterState, "reroute");
+        }
+        // Ensure all shards are started
+        while (initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.INITIALIZING).size() > 0) {
+            initialClusterState = initialClusterStrategy.applyStartedShards(
+                initialClusterState,
+                initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.INITIALIZING)
+            );
+        }
+
+        assert (initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.STARTED).size() == totalShardCount);
+        assert (initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.INITIALIZING).size() == 0);
+        assert (initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.RELOCATING).size() == 0);
+        // make sure shards are only allocated on tag1
+        for (ShardRouting startedShard : initialClusterState.getRoutingNodes().shardsWithState(ShardRoutingState.STARTED)) {
+            assert (initialClusterState.getRoutingNodes().node(startedShard.currentNodeId()).node().getAttributes().get("tag")).equals(
+                "tag_1"
+            );
+        }
     }
 
     private int toInt(String v) {
         return Integer.valueOf(v.trim());
     }
 
     @Benchmark
-    public ClusterState measureAllocation() {
+    public ClusterState measureExclusionOnZoneAwareStartedShard() throws Exception {
```

Comment:
Can we pls add javadoc for the new APIs here.

Reply:
This is JMH benchmark I guess, not sure which API

Reply:
I guess by API, you probably means class method. Benchmark are like tests and IMO, should be fine for test methods to not have documentation?

Reply:
Yes, I meant the class methods. The main objective was to make the test understandable for other developers contributing to the project. Although arguably tests may not need javadocs because the test name is usually descriptive enough, but for some complex scenarios, it can help IMO. The javadocs we publish [here](https://opensearch.org/javadocs/beta-1/OpenSearch/benchmarks/build/docs/javadoc/org/opensearch/benchmark/routing/allocation/AllocationBenchmark.html) fail to capture any details. Just fyi, we are also [adding missingJavadoc checks](https://github.com/opensearch-project/OpenSearch/pull/715) (though not strictly enabled yet), and the benchmarks module gets captured as a failure

```
/OpenSearch/benchmarks/src/main/java/org/opensearch/benchmark/routing/allocation/AllocationBenchmark.java:173: error - AllocationBenchmark.measureAllocation (method): javadocs are missing
```

That being said, I understand it is arguable and can be a personal preference. I had to dive into code to understand what the test actually does, so thought having some documentation from the author might be helpful.

LMK what you think. If you think we don't need javadoc for `benchmarks` module, we can open a discuss issue and close it separately. Approving this PR nevertheless.

Reply:
I have created [separate issue](https://github.com/opensearch-project/OpenSearch/issues/724), to discuss documentation for all benchmarks and not just this one in general